src/compiler/types.ts

```diff
@@ -2638,11 +2639,11 @@ namespace ts {
     }
 
     /**
+     * Represents @augments @extends and @implements
      * Note that `@extends` is a synonym of `@augments`.
-     * Both tags are represented by this interface.
      */
-    export interface JSDocAugmentsTag extends JSDocTag {
-        kind: SyntaxKind.JSDocAugmentsTag;
+    export interface JSDocHeritageTag extends JSDocTag {

```

Comment:
this is a little surprising. I expected to see a new type JSDocImplementsTag and then maybe
`type JSDocHeritageTag = JSDocAugmentsTag | JSDocImplementsTag` for convenience.

Reply:
I could create two separate types, but they happen to have the exact same structure, since `@implements` in Jsdoc takes one type, but can appear multiple times (at least as far as I have read and seen). I thought it made sense to unify all of them since they kind of are related.

Reply:
Hmmm, I still think it would be less confusing to have two separate types since that's how the other types work.

Also: good point, I didn't think about multiple `@implements`. Did you have tests for that already?

Reply:
Ok, I will split up the types. I did do some testes and multiple `@implement` tags work, but I did not add the tests to git yet, I will do so this weekend.